### PR TITLE
Remove chunk event subscription when disposing a MultiFrameImageStreamCompleter

### LINKED ISSUE #77576: Exception has occurred when call ImageProvider.evict at state.dispose
I want to remove image data from memory when image  item is dispose.

## Steps to Reproduce
1.run demo, scroll list

``` dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, this.title}) : super(key: key);
  final String? title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  List<String> images = <String>[
    'https://photo.tuchong.com/4870004/f/298584322.jpg',
    'https://photo.tuchong.com/16389644/f/1303840119.jpg',
    'https://photo.tuchong.com/1924454/f/153280211.jpg',
    'https://photo.tuchong.com/1824846/f/601739531.jpg',
    'https://photo.tuchong.com/2353448/f/147183522.jpg',
    'https://photo.tuchong.com/2732194/f/514055506.jpg',
    'https://photo.tuchong.com/19454111/f/940639122.jpg',
    'https://photo.tuchong.com/12772247/f/950208723.jpg',
    'https://photo.tuchong.com/16389644/f/1303840119.jpg',
    'https://photo.tuchong.com/1924454/f/153280211.jpg',
    'https://photo.tuchong.com/1824846/f/601739531.jpg',
    'https://photo.tuchong.com/2353448/f/147183522.jpg',
    'https://photo.tuchong.com/2732194/f/514055506.jpg',
    'https://photo.tuchong.com/19454111/f/940639122.jpg',
    'https://photo.tuchong.com/12772247/f/950208723.jpg',
    'https://photo.tuchong.com/16389644/f/1303840119.jpg',
    'https://photo.tuchong.com/1924454/f/153280211.jpg',
    'https://photo.tuchong.com/1824846/f/601739531.jpg',
    'https://photo.tuchong.com/2353448/f/147183522.jpg',
    'https://photo.tuchong.com/2732194/f/514055506.jpg',
    'https://photo.tuchong.com/19454111/f/940639122.jpg',
    'https://photo.tuchong.com/12772247/f/950208723.jpg',
    'https://photo.tuchong.com/1824846/f/601739531.jpg',
    'https://photo.tuchong.com/20789574/f/1191315286.jpg',
    'https://photo.tuchong.com/2353448/f/147183522.jpg',
    'https://photo.tuchong.com/15809857/f/158194644.jpg',
    'https://photo.tuchong.com/16389644/f/1303840119.jpg',
    'https://photo.tuchong.com/1924454/f/153280211.jpg',
    'https://photo.tuchong.com/3193193/f/1176439059.jpg',
    'https://photo.tuchong.com/2353448/f/147183522.jpg',
    'https://photo.tuchong.com/2732194/f/514055506.jpg',
    'https://photo.tuchong.com/15809857/f/158194644.jpg',
    'https://photo.tuchong.com/12772247/f/950208723.jpg',
    'https://photo.tuchong.com/2992739/f/226216625.jpg',
    'https://photo.tuchong.com/1824846/f/601739531.jpg',
    'https://photo.tuchong.com/20789574/f/1191315286.jpg',
    'https://photo.tuchong.com/1755665/f/206885901.jpg',
    'https://photo.tuchong.com/16389644/f/673974708.jpg',
    'https://photo.tuchong.com/2353448/f/973145661.jpg',
    'https://photo.tuchong.com/19454111/f/940639122.jpg',
    'https://photo.tuchong.com/15809857/f/920060890.jpg',
    'https://photo.tuchong.com/2742991/f/587652854.jpg',
    'https://photo.tuchong.com/239828/f/350019176.jpg',
    'https://photo.tuchong.com/19702103/f/679282943.jpg',
    'https://photo.tuchong.com/2461600/f/813958177.jpg',
    'https://photo.tuchong.com/475156/f/560914063.jpg',
    'https://photo.tuchong.com/19750051/f/1264584179.jpg',
  ];
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title!),
      ),
      body: ListView.builder(
        itemBuilder: (context, index) {
          return ImageItem(images[index]);
        },
        itemCount: images.length,
      ),
    );
  }
}

class ImageItem extends StatefulWidget {
  ImageItem(this.uri);
  final String uri;
  @override
  _ImageItemState createState() => _ImageItemState();
}

class _ImageItemState extends State<ImageItem> {
  late NetworkImage imageProvider;
  @override
  void initState() {
    super.initState();
    imageProvider = NetworkImage(widget.uri);
  }

  @override
  void dispose() {
    imageProvider.evict();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Image(
      image: imageProvider,
    );
  }
}


```

## Exception has occurred.

```
Exception has occurred.
_AssertionError ('package:flutter/src/painting/image_stream.dart': Failed assertion: line 326 pos 12: '_completer == null': is not true.)

_AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:46)
_AssertionError._throwNew (dart:core-patch/errors_patch.dart:36)
ImageStream.setCompleter (/Users/zmtzawqlp/Documents/flutter/stable/packages/flutter/lib/src/painting/image_stream.dart:326)
ImageProvider.resolve.<anonymous closure> (/Users/zmtzawqlp/Documents/flutter/stable/packages/flutter/lib/src/painting/image_provider.dart:337)
<asynchronous gap> (Unknown Source:0)
```

This is not found in v1.22.6

```
[✓] Flutter (Channel stable, 2.0.0, on macOS 11.0.1 20B29 darwin-x64, locale
    zh-Hans-CN)
    • Flutter version 2.0.0 at /Users/zmtzawqlp/Documents/flutter/stable
    • Framework revision 60bd88df91 (5 days ago), 2021-03-03 09:13:17 -0800
    • Engine revision 40441def69
    • Dart version 2.12.0
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/zmtzawqlp/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/zmtzawqlp/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.54.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (3 available)
    • iPhone 12 Pro Max (mobile) • 2204B20F-9E1B-4872-9A94-A9A734D2BC02 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)            • macos                                •
      darwin-x64     • macOS 11.0.1 20B29 darwin-x64
    • Chrome (web)               • chrome                               •
      web-javascript • Google Chrome 88.0.4324.192
```

### PR DESCRIPTION
**Pull request by dnfield**
Otherwise, when data comes in from the network, we might hit a code path we expect not to hit when we're disposed.

Also deletes a commented out test.

Fixes https://github.com/flutter/flutter/issues/77576

### CODE REVIEW

File: packages/flutter/test/painting/image_stream_test.dart [hunk 1]
```diff
@@ -811,44 +811,31 @@ void main() {
     expect(oneFrameCodec.numFramesAsked, 1);
   });  // https://github.com/flutter/flutter/issues/82532
 
-  // TODO(amirh): enable this once WidgetTester supports flushTimers.
```
Comment (Piinks) [inline packages/flutter/test/painting/image_stream_test.dart:814]: https://github.com/flutter/flutter/issues/30344 and its duplicate https://github.com/flutter/flutter/issues/24166 are both closed. Synced with @dnfield offline to see if this test actually works

Comment (dnfield) [reply packages/flutter/test/painting/image_stream_test.dart:814]: Test does not work, I'm not even really sure what it was supposed to test, and multiple things it referred to have been refactored. We've made it this far without this test so we're probably ok.

Comment (Piinks) [reply packages/flutter/test/painting/image_stream_test.dart:814]: Got rot. Thanks for checking!

**Review by Piinks — APPROVED** [review]
LGTM!

**Pull request merged**